base/db/R/clone_pft.R

```diff
@@ -0,0 +1,105 @@
+##' Duplicate existing pft with associated priors, species, and cultivars
+##'
+##' Creates a new pft that is a duplicate of an existing pft,
+##' including relationships with priors, species, and cultivars (if any) of the existing pft.
+##' This function mimics the 'clone pft' button in the PFTs record view page in the 
+##' BETYdb web interface for PFTs that aggregate >=1 species, but adds the ability to 
+##' clone the cultivar associations.
+##'
+##' @param parent.pft.name name of PFT to duplicate
+##' @param new.pft.name name for new PFT. Must not be the same as parent.pft.name
+##' @param new.pft.definition text for the new PFT's definition field.
+##' @param settings PEcAn settings list, used only for BETYdb connection parameters
+##' @return ID of the newly created pft in database, creates new PFT as a side effect
+##' @author David LeBauer, Chris Black
+##' @examples \dontrun{
+##' clone_pft(parent.pft.name    = "tempdecid",
+##'           new.pft.name       = "mytempdecid",
+##'           new.pft.definition = "mytempdecid is a new pft",
+##'           settings = pecan_settings_list)
+##' }
+clone_pft <- function(parent.pft.name,
+                      new.pft.name,
+                      new.pft.definition,
+                      settings){
+
+  if (new.pft.name == parent.pft.name) {
+    PEcAn.logger::logger.severe("new.pft.name must not be the same as parent.pft.name")
+  }
+
+  if (length(new.pft.name) > 1 || length(parent.pft.name) > 1) {
+    PEcAn.logger::logger.severe("multiple PFT names given, and clone_pft only knows how to handle one at a time.")
+  }
+
+  con <- db.open(settings$database$bety)
+  on.exit(db.close(con))
+
+  parent.pft <- (dplyr::tbl(con, "pfts")
+    %>% dplyr::filter(name == parent.pft.name)
+    %>% dplyr::collect())
```

Comment:
No need to change, because this works and it's perfectly clear what's going on...but why wrap these pipes in parentheses? I don't think I've ever seen anyone do this. Just curious.

Reply:
In multiline expressions I strongly prefer to have infix operators at the left end of the line, so the parens are just to make sure the whole pipeline is evaluated as a unit. Do we have a formal style standard about this? If not I'll argue that we should adopt my way :)

Reply:
In lieu of formal style we should defer to convention. And PEcAn style should defer to the broader community conventions by default - unless there is a compelling reason to do otherwise.

Reply:
To the extent that we follow the [Tidyverse style guide](http://style.tidyverse.org/pipes.html), the style is this:

```r
rabbit %>%
  hop() %>%
  jump()
```

That said, I have no desire to be dogmatic about it -- I think other aspects of coding style are much more important, and the code is perfectly clear and readable as is.